uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/api/v3/ContactTest.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+use CRM_HRCore_Test_Fabricator_Contact as ContactFabricator;
+
+
+/**
+ * Class api_v3_LeaveRequestTest
+ *
+ * @group headless
+ */
+class api_v3_ContactTest extends BaseHeadlessTest {
+
+  use CRM_HRLeaveAndAbsences_LeaveRequestHelpersTrait;
+  use CRM_HRLeaveAndAbsences_LeaveBalanceChangeHelpersTrait;
+  use CRM_HRLeaveAndAbsences_LeaveManagerHelpersTrait;
+  use CRM_HRLeaveAndAbsences_SessionHelpersTrait;
+
+  private $manager;
+
+  private $contact1;
+
+  private $contact2;
+
+  public function setUp() {
+    CRM_Core_DAO::executeQuery('SET foreign_key_checks = 0;');
+
+    $this->contact1 = ContactFabricator::fabricate();
+    $this->contact2 = ContactFabricator::fabricate();
+    $this->manager = ContactFabricator::fabricate();
+    $this->registerCurrentLoggedInContactInSession($this->manager['id']);
+
+    $this->setLeaveApproverRelationshipTypes([
+      'approves leaves for',
+      'manages things for',
+    ]);
+  }
+
```

Comment:
Can we have a test where the manager only manages some of the contacts and we assert that only the ones managed by the current logged in user are returned?

Another variation of this could be a tests were we have 2 different managers and we assert that the API only returns the managees of the current logged in manager